src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -20,145 +16,190 @@
+						new Translation2d(
+								Units.inchesToMeters(12.375), Units.inchesToMeters(10.375)), // front left
+						new Translation2d(
+								Units.inchesToMeters(12.375), Units.inchesToMeters(-10.375)), // front right
+						new Translation2d(
+								Units.inchesToMeters(-12.375), Units.inchesToMeters(10.375)), // back left
+						new Translation2d(
+								Units.inchesToMeters(-12.375), Units.inchesToMeters(-10.375)) // back right
+					}
+					: new Translation2d[] {
+						new Translation2d(Units.inchesToMeters(8.5), Units.inchesToMeters(8.5)), // front left
+						new Translation2d(Units.inchesToMeters(8.5), Units.inchesToMeters(-8.5)), // front right
+						new Translation2d(Units.inchesToMeters(-8.5), Units.inchesToMeters(8.5)), // back left
+						new Translation2d(Units.inchesToMeters(-8.5), Units.inchesToMeters(-8.5)) // back right
+					};
 
 	SwerveDriveKinematics kinematics =
 			new SwerveDriveKinematics(
 					moduleLocations[0], moduleLocations[1], moduleLocations[2], moduleLocations[3]);
 
-	private AHRS gyroscope;
+	private Gyroscope gyroscope;
 
 	private SwerveDriveOdometry odometry;
 	private Pose2d pose;
 
 	private Field2d field = new Field2d();
 
 	public DrivebaseSubsystem() {
-		gyroscope = new AHRS(SerialPort.Port.kMXP);
+		gyroscope = IS_COMP ? new Pigeon2Gyro(Hardware.GYRO_PORT) : new NavXGyro(SerialPort.Port.kMXP);
 
-		odometry =
-				new SwerveDriveOdometry(
-						kinematics, Rotation2d.fromDegrees(gyroscope.getYaw()), getModulePositions());
+		odometry = new SwerveDriveOdometry(kinematics, gyroscope.getAngle(), getModulePositions());
 		pose = odometry.getPoseMeters();
 
 		balanceController =
-				PFFController.ofDouble(tipF, tipP)
-						.setTargetPosition((double) gyroscope.getRoll())
-						.setTargetPositionTolerance(tipTolerance);
+				PFFController.ofDouble(TIP_F, TIP_P)
+						.setTargetPosition(gyroscope.getRawRoll().getDegrees())
+						.setTargetPositionTolerance(TIP_TOLERANCE);
 
 		// configure encoders offsets
 		for (int i = 0; i < moduleEncoders.length; i++) {
 			moduleEncoders[i].configFactoryDefault();
+			moduleEncoders[i].configSensorInitializationStrategy(
+					SensorInitializationStrategy.BootToAbsolutePosition);
 		}
 
 		// configure drive motors
 		for (int i = 0; i < moduleDriveMotors.length; i++) {
-			WPI_TalonFX driveMotor = moduleDriveMotors[i];
-			driveMotor.setNeutralMode(NeutralMode.Brake);
-			driveMotor.setSensorPhase(true);
-
-			driveMotor.configNeutralDeadband(0.01);
-			driveMotor.configSelectedFeedbackSensor(
-					TalonFXFeedbackDevice.IntegratedSensor, talonFXLoopNumber, canTimeoutMS);
-
-			driveMotor.config_kP(talonFXLoopNumber, 0.1);
-			driveMotor.config_kI(talonFXLoopNumber, 0.001);
-			driveMotor.config_kD(talonFXLoopNumber, 1023.0 / 20660.0);
+			MotorController driveMotor = moduleDriveMotors[i];
+			driveMotor.setNeutralMode(MotorController.MotorNeutralMode.BRAKE);
+
+			if (IS_COMP) {
+				driveMotor.setControlMode(MotorControlMode.VOLTAGE);
```

Comment:
Really ought to use velocity control to get the bot to drive straight, so I'd highly recommend re-enabling and then tuning that (REV has a sample [here](https://github.com/REVrobotics/SPARK-MAX-Examples/blob/master/Java/Velocity%20Closed%20Loop%20Control/src/main/java/frc/robot/Robot.java)). There are hacks you can do in the drivebase subsystem to partially correct steering errors from the motors moving different speeds - but it'd be better to just get them moving at the right speed in the first place. Note that [the motor controller PIDF runs at 1kHz](https://docs.revrobotics.com/sparkmax/operating-modes/closed-loop-control) which is way faster than any control loop you can do in the Rio-hosted robot code here.

Reply:
Not blocking for this PR to get things kindof working, but needs to be fixed before auto pathing really gets underway.